kolibri/core/context_processors/custom_context_processor.py

```diff
@@ -33,12 +33,10 @@ def pass_browser_entry(agent, entry):
 
 
 def supported_browser(request):
-    if 'supported_browser' not in request.session:
-
-        user_agent = user_agents.parse(request.META.get('HTTP_USER_AGENT', ''))
-        request.session['supported_browser'] = all(
-            pass_browser_entry(user_agent, entry) for entry in browser_requirements)
-    return {'supported_browser': request.session['supported_browser']}
+    user_agent = user_agents.parse(request.META.get('HTTP_USER_AGENT', ''))
+    return {
+        'supported_browser': all(pass_browser_entry(user_agent, entry) for entry in browser_requirements)
```

Comment:
I see the returned value is being removed from the session. 
wouldn't it be good to use the session to cache the value and return it if it already exixts? it's being called/recalculated from the base.html template whenever the SPA changes.

Reply:
I can either cache the pages, or I can use the session object. A middle ground might be to memoize the calculation based on user agent?

Reply:
Yeah, I hadn't thought of using the user agent, but your idea of memoizing based on it would be the more efficient solution.